onchain/src/social/social.cairo

```diff
@@ -0,0 +1,64 @@
+use super::request::SocialRequest;
+use super::transfer::Transfer;
+
+#[starknet::interface]
+pub trait ISocial<TState> {
+    fn handle_transfer(ref self: TState, request: SocialRequest<Transfer>);
+}
+
+#[starknet::component]
+pub mod SocialComponent {
+    use openzeppelin::token::erc20::interface::{ERC20ABIDispatcher, ERC20ABIDispatcherTrait};
+    use super::super::account::{ISocialAccountDispatcher, ISocialAccountDispatcherTrait};
+    use super::super::request::{SocialRequest, SocialRequestImpl};
+    use super::super::transfer::Transfer;
+
+    #[storage]
+    struct Storage {
+        public_key: u256,
```

Comment:
`public_key` should not be in Social component storage.

Reply:
You should either add a constraint on `TContractState` (like `HasPublicKey` defined in the main contract, not sure if this will work though) or define public key in a separate component and component dependencies: https://book.cairo-lang.org/ch16-02-02-component-dependencies.html.